### DISCUSSION TITLE
tsx -r dotenv/config scripts/ingest-data.ts   - issue

### DISCUSSION THREAD

**Discussion by anujcb** [mayooear/ai-pdf-chatbot-langchain#216] [Q&A] [answered]
I am getting the following error. New to node.js so not sure if i am missing any packages. I verified that I have a node_modules folder and a yarn folder inside it 

C:\Users\xyz\AppData\Roaming\npm\node_modules\yarn
running v12.14.0 for node
i ahve updated the .env files with the api key

any help is appreciated.

C:\Users\xyz\gpt4-pdf-chatbot-langchain>npm run ingest

> gpt4-langchain-pdf-chatbot@0.1.0 ingest C:\Users\Anil\gpt4-pdf-chatbot-langchain
> tsx -r dotenv/config scripts/ingest-data.ts

'tsx' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gpt4-langchain-pdf-chatbot@0.1.0 ingest: `tsx -r dotenv/config scripts/ingest-data.ts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gpt4-langchain-pdf-chatbot@0.1.0 ingest script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

**Comment by hugobatista** [comment]
check node -v
you should be running >18

**Reply by anujcb** [reply]
I upgraded to 18, still the same issue

C:\Users\xyz\gpt4-pdf-chatbot-langchain>node -v
v18.16.0

C:\Users\xyz\gpt4-pdf-chatbot-langchain>npm -v
9.6.5

C:\Users\xyz\gpt4-pdf-chatbot-langchain>npm run ingest

> gpt4-langchain-pdf-chatbot@0.1.0 ingest
> tsx -r dotenv/config scripts/ingest-data.ts

'tsx' is not recognized as an internal or external command,
operable program or batch file.

**Reply by hugobatista** [reply, marked as answer]
did you run "yarn install"?

**Reply by anujcb** [reply]
awesome! its working, I was installing the yarn in the wrong dir, when i installed it from C:\Users\xyz\gpt4-pdf-chatbot-langchain it worked!! Thank you!
